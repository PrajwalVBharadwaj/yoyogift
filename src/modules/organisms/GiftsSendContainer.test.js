import React from "react";
import { shallow, userDetails, cardDetails } from "../../Enzyme";
import { GiftsSendContainer } from "./GiftsSendContainer";

describe("Testing GiftsSendContainer", () => {
  it("Snapshot testing", () => {
    const wrapper = shallow(
      <GiftsSendContainer
        isLoggedIn={true}
        user={userDetails}
        fetchSentCards={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Testing for sent cards", () => {
    const wrapper = shallow(
      <GiftsSendContainer isLoggedIn={true} sentCards={[cardDetails]} />
    );
    expect(wrapper.find("div").length).toBe(1);
  });

  it("Testing for no sent cards", () => {
    const wrapper = shallow(
      <GiftsSendContainer isLoggedIn={true} sentCards={[]} />
    );
    expect(wrapper.find("h2").length).toBe(1);
  });

  it("Testing for not logged in", () => {
    const wrapper = shallow(<GiftsSendContainer isLoggedIn={false} />);
    expect(wrapper.find("Redirect").length).toBe(1);
  });
});
