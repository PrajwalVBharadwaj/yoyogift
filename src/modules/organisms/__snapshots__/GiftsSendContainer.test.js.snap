// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GiftsSendContainer Snapshot testing 1`] = `ShallowWrapper {}`;
